packages/common/test/tools/queue/TaskQueue.test.ts

```diff
@@ -136,8 +146,13 @@ describe(TaskQueue.name, () => {
     queue.addToBack(1)
     queue.addToBack(2)
 
+    await time.runAllAsync()
+
     await queue.waitTilEmpty()
 
     expect(queue.length).toEqual(0)
   })
 })
+
+const wait = (delay: number) =>
```

Comment:
Unfortunately `fake-timers` do not work well with `timers/promises`, so I had to make our own promisified version.

Reply:
this should be in-code comment